app/models/viewmodels/notificationspage/ControlResponseConverter.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.viewmodels.notificationspage
+
+import java.time.format.DateTimeFormatter
+import java.time.{Instant, ZoneId}
+
+import javax.inject.{Inject, Singleton}
+import models.notifications.NotificationFrontendModel
+import models.notifications.ResponseType.ControlResponse
+import models.viewmodels.decoder.Decoder
+import play.api.Logger
+import play.api.i18n.Messages
+import play.twirl.api.Html
+
+@Singleton
+private[notificationspage] class ControlResponseConverter @Inject()(decoder: Decoder)
```

Comment:
why package private ?

Reply:
Because I don't expect it to be used outside. These are actually private objects to the Factory in some sense. Their content is the Factory's implementation which would be private otherwise.